### ISSUE TITLE
Upgrade actix version to 0.9

### ISSUE THREAD

**Issue by oronsh** [railgun-rs/actix-raft#40]
Upgrade actix to version >= 0.9 so applications using actix 0.9 could use actix-raft

**Comment by thedodd** [comment]
Good call. Will take care of this ASAP!

**Comment by oronsh** [comment]
Great, started working on it myself but frankly I do not understand well enough how to refactor actor futures to async await syntax. After you will have a branch so I can see some examples of converting those chains of actor futures I would love to join! :)

**Comment by thedodd** [comment]
Hmmm. Actix 0.9 does not itself support async/await syntax for the actor system quite yet. It uses tokio 0.2 and std::futures, but it still heavily uses the ActorFuture abstraction which does not currently play nicely with async/await at this point. Definitely let me know if I am wrong, but I don't think I am. There is another issue opened for cutting over to async/await when Actix is ready (#38).

The update to 0.9 should be pretty simple.

**Comment by oronsh** [comment]
When I tried to upgrade to actix 0.9 in my code I've noticed a lot of errors about "method not found" for some combinators like `and_then` or `map_err` in `ActorFuture` which code that was written with 0.8.3 is heavily relied on. https://docs.rs/actix/0.9.0/actix/fut/trait.ActorFuture.html
Hope that doesn't make things difficult.

Anyway I thought the purpose of ActorFuture is to pass the actor and context references to the future combinators but I guess that if you use async/await you can just use the `.send` future of actix alone without combinators.

**Comment by raultang** [comment]
Hei Guys, what is the status, the async/await feature is really helpful.

**Comment by thedodd** [comment]
Hey @raultang a lot of the discussion which I am waiting on is here: https://github.com/actix/actix/issues/308. However, I'm not sure how long it will take for that discussion to resolve & for async/await support to land within actix itself.

As such, I will do my best to at least update actix-raft to the latest version of actix soon. It will not be a 1.0 release, as I do want to wait until async/await is supported internally in actix & update this code to use that pattern before issuing a 1.0 release.
